mountpoint-s3/src/inode.rs

```diff
@@ -351,6 +351,59 @@ impl Superblock {
 
         Ok(LookedUp { inode, stat })
     }
+
+    /// Create a new directory inode
+    pub async fn createdir<OC: ObjectClient>(
```

Comment:
This is copied from `create`, with the only changes in `kind`/`kind_data`. Planning to refactor if approach confirmed.